src/deps-index.ts

```diff
@@ -13,12 +13,14 @@
  */
 import {AssertionError} from 'assert';
 import {Analyzer} from 'polymer-analyzer';
+import * as urlLib from 'url';
 
+import * as urlUtils from './url-utils';
 import {UrlString} from './url-utils';
 
 export interface DepsIndex {
   // An index of entrypoint -> html dependencies
-  entrypointToDeps: Map<string, Set<string>>;
+  entrypointToDeps: Map<UrlString, Set<UrlString>>;
```

Comment:
Buys nothing with type checking.  I guess I liked the idea of them documenting the intent/content of the strings, but I haven't found a way to leverage usefully.  I'm open to trading for `string` at some point.